my-dsl.md

```diff
@@ -2,37 +2,46 @@
 _What is the name of the language? Link the name to its webpage 
 (if appropriate)._
 
+The name of the language is CHEM. It is fairly old, so it does not have a webpage. However, a paper that talks about it can be found at this link: http://dl.acm.org/citation.cfm?id=315691
 
 # Domain
 _Describe the language's domain in five words._
 
+Creates chemical structure diagrams.
 
 # Computational model
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of 
 what happens when a program in your DSL runs._
 
+When a program runs, it reads lines and draws something onto the canvas based on what was on the line. It can draw things like benzene and can connect them with things like double bonds. The final output is a chemical structure diagram.
 
 # DSL-ness
 _Fowler writes about a spectrum of languages, from general-purpose languages to 
 "purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
 and why?_ 
 
+CHEM is purely domain-specific. It focuses on drawing chemical structure diagrams and does not support drawing other diagrams. It really cannot do anything else besides draw chemical structure diagrams. Valid code in CHEM is something like "doublebond -135 from Rl.V3 ; 0" and cannot diverge from this kind of writing.
 
 # Internal or external?
 _Is the language implemented as an internal or external DSL? 
 Justify your answer._
 
+One could argue that this is an external DSL because it is meant to be used stand alone. It also looks like an internal DSL because it takes some operations from the PIC programming language and extends it slightly to make it nice for chemists to use. However, internal DSLs are a subset of their host language, and the host language is typically a general-purpose language (PIC is a DSL). So I would say that CHEM is an external DSL.
```

Comment:
I thought this distinction came more from whether it had its own compiler than how much it feels like its own language.  I would disagree with the argument that one can classify a DSL as external because it is meant to be used stand alone.  I think that is the difference between an API and an internal DSL.